File: app/src/main/java/com/confessionsearchapptest/release1/data/notes/NoteDao.java
package com.confessionsearchapptest.release1.data.notes;

import androidx.lifecycle.LiveData;
import androidx.room.*;

import java.util.List;

//Thanks to Mitch at SQLite Learning for providing code to help set up this interface here
@Dao
public interface NoteDao {

    @Insert
    long[] insertNotes(Notes... notes);
    @Query("SELECT * FROM notes")
    LiveData<List<Notes>> getNotes();
    @Delete
    int delete(Notes... notes);
    @Update
    int updateNotes(Notes... notes);
}

File: app/src/main/java/com/confessionsearchapptest/release1/data/notes/NoteDao.kt
package com.confessionsearchapptest.release1.data.notes

import androidx.lifecycle.LiveData
import androidx.room.*


@Dao
interface NoteDao {
    @Insert
    fun insertNotes(vararg notes: Notes?): LongArray?

    @Query("SELECT * FROM notes")
    fun fetchNotes(): LiveData<List<Notes?>?>?

    @Delete
    fun delete(vararg notes: Notes?): Int

    @Update
    fun updateNotes(vararg notes: Notes?): Int
}